tests/unit_tests/plugin_registries/test_plugin_registries.py

```diff
@@ -10,27 +10,214 @@
+        """Test Plugin Registry interfaces validity."""
+        yaml_interfaces = []
+        module_interfaces = []
+
+        # NOTE: all <pkg>/interfaces/__init__.py files MUST
+        # contain a "module_based_interfaces" list and a
+        # "yaml_based_interfaces" list - this is how we
+        # identify the valid interface names.
+        # We must avoid actually importing the interfaces within
+        # the geoips repo - or we will end up with a circular import
+        # due to BaseInterface.
+        for pkg in get_entry_point_group("geoips.plugin_packages"):
+            try:
+                mod = import_module(f"{pkg.value}.interfaces")
+            except ModuleNotFoundError as resp:
+                if f"No module named '{pkg.value}.interfaces'" in str(resp):
+                    continue
+                else:
+                    raise ModuleNotFoundError(resp)
+            module_interfaces += mod.module_based_interfaces
+            yaml_interfaces += mod.yaml_based_interfaces
+
+        bad_interfaces = []
+        for plugin_type in ["module_based", "yaml_based"]:
+            for interface in current_registry[plugin_type]:
+                if (
+                    interface not in module_interfaces
+                    and interface not in yaml_interfaces
+                ):
+                    error_str = f"\nPlugin type '{plugin_type}' does not allow "
+                    error_str += f"interface '{interface}'.\n"
+                    error_str += "\nValid interfaces: "
+                    if plugin_type == "module_based":
+                        interface_list = module_interfaces
+                    else:
+                        interface_list = yaml_interfaces
+                    error_str += f"\n{interface_list}\n"
+                    error_str += "\nPlease update the following plugins "
+                    error_str += "to use a valid interface:\n"
+                    error_str += pformat(current_registry[plugin_type][interface])
+                    bad_interfaces.append(error_str)
+        if bad_interfaces:
+            error_str = "The following interfaces were not valid:\n"
+            for error in bad_interfaces:
+                error_str += error
+            raise PluginRegistryError(error_str)
+
+
+class TestPluginRegistry:
+    """
+    Pytest-based Unit Test for the PluginRegistry Class.
+
+    Note: Since we are not able to initlialize this class due to restrictions placed by
+    Pytest, if you want to change the test files used, simply replace the location below
+    with the location of your new test files.
+    """
+
+    _default_fpaths = glob(
         str(__file__).replace("test_plugin_registries.py", "files/**/*.yaml"),
         recursive=True,
     )
 
-    plugin_registry = PluginRegistry(fpaths)
+    _pr_validator = PluginRegistryValidator(_default_fpaths)
+
+    # Couldn't implement this class via inheritance because PyTest Raised this error:
```

Comment:
If including a comment like this, we should explain what the error means, and why it's fine

Reply:
I think Evan is saying that he isn't subclassing in order to avoid having this error occur. Rather than inheriting from `PluginRegistry`, he is setting it to the class attribute `plugin_registry`.

Reply:
Added a comment explaining this more.